lib/AST/Expr.cpp

```diff
@@ -2287,9 +2287,8 @@ TypeExpr *TypeExpr::createForSpecializedDecl(DeclRefTypeRepr *ParentTR,
           return nullptr;
       }
 
-      if (auto *identBase =
-              dyn_cast<IdentTypeRepr>(memberTR->getBaseComponent())) {
-        if (isUnboundGenericComponent(identBase))
+      if (auto *identRoot = dyn_cast<IdentTypeRepr>(memberTR->getRoot())) {
```

Comment:
Are all the usages of getRoot() correct? Maybe some of them should be getBase()?